timlee/networking-with-a-raspberry/README.md

```diff
@@ -0,0 +1,280 @@
+# Networking with a Raspberry
+
+![Raspberry Pi logo](./raspberry-pi-logo.webp)
+
+_[Tim Lee](../) — 21st January 2021_
+
+Want to learn about networking, proxies and web servers? Read on.
+
+## Introduction
+
+As a developer, I have always felt safest _in the code_. For a long time, though, I assumed that "network stuff" involving servers and connections was complicated and unnecessary for me to know.
+
+As the lines between the classically separate roles of 'software developer' and 'operations engineer' started to blend together into 'DevOps', I realised there was no escaping it.
+
+Luckily, servers are nothing to be afraid of - in fact, they're a handy, multi-bladed Swiss army knife that can be used for all sorts of things. In this post, I'll be demonstrating some of the most common use-cases with the aid of the humble (but mighty) Raspberry Pi.
+
+### What this post covers
+
+Let's pretend you're in the office of your company headquarters, Badgers 4 U, because offices are still a thing that people go to. You want to set up a really cool internal site so that people can find important (but not public) information about their favourite types of Badger.
+
+To do that we'll need to cover off a few things:
+
+1. [Setting up a web server](#step-1-web-server)
+2. [Configuring DNS](#step-2-dns)
+3. [Creating SSL certificates](#step-3-SSL-certificates)
+4. [Using a reverse proxy](#step-4-reverse-proxy)
+
+### What you will need
+
+A Raspberry Pi (I used a [Raspberry Pi 4b](https://www.raspberrypi.org/products/raspberry-pi-4-model-b/)) connected to your local WiFi network. However, if you don't have a Pi to hand, you can use pretty much any other machine on your network.
+
+![Raspberry Pi circuit illustration](./raspberry-pi.svg)
+
+## Step 1: Web server
+
+If you want people to be able to access your site then you need to have a way of providing it to them. For this you need a web server. This will store your site and hand it out to anyone who comes asking for it.
+
+### What is a web server?
+
+A web server is just some software running on a networked machine. It can be dumb, like a file server used to serve static documents (might sound familiar if you've ever used an AWS S3 bucket) or a bit smarter, interpreting requests and doing some processing to dynamically render web pages (termed server-side rendering) or providing results to API requests (for example, a GraphQL API running on a NodeJS server).
```

Comment:
I would suggest instead of just naming AWS S3 bucket, just say:

as CDN such as  AWS S3 bucket, Google Cloud Storage, etc.